src/floating_label.cpp

```diff
@@ -177,29 +180,47 @@ void floating_label::draw(surface screen)
 		}
 	}
 
-	SDL_Rect rect = sdl::create_rect(xpos(surf_->w), ypos_, surf_->w, surf_->h);
+	SDL_Point pos = get_loc(time);
+	buf_pos_ = sdl::create_rect(pos.x, pos.y, surf_->w, surf_->h);
 	const clip_rect_setter clip_setter(screen, &clip_rect_);
-	sdl_copy_portion(screen,&rect,buf_,nullptr);
-	sdl_blit(surf_,nullptr,screen,&rect);
+	sdl_copy_portion(screen, &buf_pos_, buf_, nullptr);
+	sdl_blit(get_surface(time), nullptr, screen, &buf_pos_);
```

Comment:
Inconsistent checking for `surf_ != nullptr` - it's not checked here, but inside `get_surface` there's a check which will make that function return nullptr.

Reply:
but a few lines above this function checks for for surf_ == nullptr